pkgs/development/compilers/ghc/8.4.2.nix

```diff
@@ -22,7 +22,7 @@
 
 , # Whether to build dynamic libs for the standard library (on the target
   # platform). Static libs are always built.
-  enableShared ? true
+  enableShared ? !targetPlatform.useAndroidPrebuilt
```

Comment:
Did you ever get this to build a valid .apk? I think we need shared here for the Haskell activity stuff to link. Maybe it worked for you though?

Reply:
I believe the old stuff from `reflex-platform` used this and successfully built APKs. I did not try it with this branch though.